src/store/storeHydrator.tsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react'
+import { Provider } from 'react-redux'
+import type { Store } from 'redux'
+
+import { getPersistedState, RootState } from '@/store'
+
+export const HYDRATE = '@@HYDRATE'
+
+export const createStoreHydrator = (makeStore: () => Store<RootState>) => {
+  return (Component: any) => {
+    return class HydrationWrapper extends React.Component<any, any> {
```

Comment:
Why does it need to be a class?

Reply:
I was encountering issues with the hydration action not dispatching first. I only got it working in a class component.